epoxy-processor/src/main/java/com/airbnb/epoxy/processor/Type.kt

```diff
@@ -43,9 +43,11 @@ class Type(val xType: XType, memoizer: Memoizer) {
                     xType.isTypeOf(CharSequence::class) || xType.isTypeOf(String::class) -> StringOrCharSequence
                     xType.typeName == ClassNames.EPOXY_STRING_ATTRIBUTE_DATA -> StringAttributeData
                     // We don't care about nullability for the purposes of type checking
-                    nonNullType == memoizer.viewOnClickListenerType -> ViewClickListener
-                    nonNullType == memoizer.viewOnLongClickListenerType -> ViewLongClickListener
-                    nonNullType == memoizer.viewOnCheckChangedType -> ViewCheckedChangeListener
+                    // Note, == does not work for type comparisons when comparing types between classpath
```

Comment:
any chance you could maybe describe the bug a bit more in the PR description. Just helps me understand a bit better. I can follow that the equals check wasn't what was needed but didn't understand what the difference in the generated code looked like

Reply:
is there any test case that could be added?

Reply:
hard to test because this happens when they are in different modules